src/librustdoc/html/format.rs

```diff
@@ -1328,16 +1327,16 @@ impl clean::GenericArg {
 }
 
 crate fn display_fn(f: impl Fn(&mut fmt::Formatter<'_>) -> fmt::Result) -> impl fmt::Display {
-    WithFormatter(Cell::new(Some(f)))
+    WithFormatter(f)
 }
 
-struct WithFormatter<F: Fn(&mut fmt::Formatter<'_>) -> fmt::Result>(Cell<Option<F>>);
+struct WithFormatter<F: Fn(&mut fmt::Formatter<'_>) -> fmt::Result>(F);
 
 impl<F> fmt::Display for WithFormatter<F>
 where
     F: Fn(&mut fmt::Formatter<'_>) -> fmt::Result,
 {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        (self.0.take()).unwrap()(f)
+        self.0(f)
```

Comment:
If I understand correctly, this allows the formatted value to be printed multiple times, for the cost of eagerly collecting from iterators, for example. Are there use cases where this is required?